staging/src/k8s.io/apiserver/pkg/registry/rest/delete.go

```diff
@@ -39,6 +39,7 @@ type GarbageCollectionPolicy string
 const (
 	DeleteDependents GarbageCollectionPolicy = "DeleteDependents"
 	OrphanDependents GarbageCollectionPolicy = "OrphanDependents"
+	Unsupported      GarbageCollectionPolicy = "Unsupported"
```

Comment:
> It does because the two sets of ignores match.

Sorry, what are the "two sets of ignores"? Do you mean `shouldOrphanDependents` and `shouldDeleteDependents`?

Here's my understanding of the code and why i think the background deletion will succeed:
Say the Unsupported object is object X. Garbage collector observes X's owner is deleted, so it deletes X with [DeletePropagationBackground](https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/garbagecollector/garbagecollector.go#L389). On the server side, shouldOrphanDependents and shouldDeleteDependents will both return false, so [deletionFinalizers](https://github.com/deads2k/kubernetes/blob/312fb1e1fa198f4715598feac659f5eeffd05032/staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go) will return "false, nil", in turn the `Delete()` function deletes X [immediately](https://github.com/deads2k/kubernetes/blob/312fb1e1fa198f4715598feac659f5eeffd05032/staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go#L984).